src/winter_onboarding_2021/fleet_management_service/handlers/fleet.clj

```diff
@@ -1,24 +1,47 @@
 (ns winter-onboarding-2021.fleet-management-service.handlers.fleet
-  (:require [winter-onboarding-2021.fleet-management-service.models.fleet :as fleet-model]
-            [clojure.spec.alpha :as s]
-            [winter-onboarding-2021.fleet-management-service.specs :as specs]
+  (:require [clojure.spec.alpha :as s]
             [ring.util.response :as response]
-            [winter-onboarding-2021.fleet-management-service.utils :refer [namespace-keys flash-msg]]
-            [winter-onboarding-2021.fleet-management-service.views.fleet :as views]))
+            [winter-onboarding-2021.fleet-management-service.specs :as specs]
+            [winter-onboarding-2021.fleet-management-service.utils :as utils]
+            [winter-onboarding-2021.fleet-management-service.config :as config]
+            [winter-onboarding-2021.fleet-management-service.views.fleet :as views]
+            [winter-onboarding-2021.fleet-management-service.models.fleet :as fleet-model]))
 
 
 (defn create-fleet [{:keys [user form-params]}]
   (let [user-id (:users/id user)
         fleet-name (:name form-params)
-        fleet-data (namespace-keys :fleets {:name fleet-name
-                                      :created-by user-id})]
+        fleet-data (utils/namespace-keys :fleets {:name fleet-name
+                                            :created-by user-id})]
     (if (s/valid? ::specs/fleets fleet-data)
       (let [fleet-id (:fleets/id (fleet-model/create fleet-data))]
-        (->  (flash-msg "Fleet created successfully!" true)
+        (->  (utils/flash-msg "Fleet created successfully!" true)
              (merge (response/redirect (format "/fleets/%s" (str fleet-id))))))
-      (->  (flash-msg "Could not create fleet, try again!" false)
+      (->  (utils/flash-msg "Could not create fleet, try again!" false)
            (merge (response/redirect "/fleets/new"))))))
 
 (defn new [_]
   {:title "Create fleet"
    :content (views/create-fleet)})
+
+(defn assoc-managers [fleet]
+  (assoc fleet
+         :managers
+         (fleet-model/managers fleet)))
+
+(defn show-fleets [request]
+  (let [user-id (get-in request [:user :users/id])
+        {:keys [page]} (:params request)
+        page-size (config/get-page-size)
+        current-page (Integer/parseInt (or page "1"))
+        offset (utils/calc-offset page-size current-page)
+        fleets (fleet-model/select-fleets-for-admin user-id
+                                                    offset
+                                                    page-size)
+        fleets-count (:count (first (fleet-model/total)))
```

Comment:
Perhaps the `total` fn can call first and :count itself? Then it could actually return the total number of fleets.

Reply:
If this is done, fleets-count can probably be inlined as well